tests/run.sh

```diff
@@ -204,6 +204,12 @@ for f in $tests; do
     # Send the test file to awesome.
     cat $f | DISPLAY=$D "$AWESOME_CLIENT" 2>&1
 
+    # Kill awesome after 1 minute (e.g. with errors during test setup).
+    (sleep 60; if [ "$(ps -o comm= $awesome_pid)" = "${AWESOME##*/}" ]; then
+        echo "Killing (stale?!) awesome (PID $awesome_pid) after 10 seconds."
```

Comment:
The message here talks about 10 seconds. You might want to fix that (and perhaps move the timeout into a variable (at the top of the file?) so that this cannot happen again).